src/main/java/org/jetbrains/research/groups/ml_methods/refactoring/detection/RefactoringDetectionApplication.java

```diff
@@ -1,11 +1,151 @@
+        List<Integer> totalRefactoringsNumbersInProcessedCommits = new ArrayList<>();
+        for (RepositoryDetectedRefactorings repositoryDetectedRefactorings : detectedRefactorings) {
+            System.out.println("--------------------------------");
+            System.out.println("Project: " +
+                    ParsingUtils.getProjectName(repositoryDetectedRefactorings.getRepository()));
+            System.out.println("Branch: " + repositoryDetectedRefactorings.getBranch());
+            if (repositoryDetectedRefactorings.isSuccess()) {
+                int processedCommitsNumber =
+                        Objects.requireNonNull(repositoryDetectedRefactorings.getExecutionInfo())
+                                .getProcessedCommitsNumber();
+                RefactoringDetectionExecutionInfo executionInfo =
+                        Objects.requireNonNull(repositoryDetectedRefactorings.getExecutionInfo());
+                int numberOfDetectedRefactorings =
+                        Objects.requireNonNull(repositoryDetectedRefactorings.getDetectedRefactorings())
+                                .size();
+                successNumber++;
+                totalProcessedCommitsNumber += processedCommitsNumber;
+                totalNumberOfDetectedRefactorings += numberOfDetectedRefactorings;
+                totalRefactoringsNumbersInProcessedCommits
+                        .addAll(executionInfo.getRefactoringsNumbersInProcessedCommits());
+                System.out.println("Result: success");
+                System.out.println("Processed commits: " + processedCommitsNumber);
+                System.out.println("Processed commits with refactorings: " +
+                        executionInfo.getProcessedCommitsWithRefactoringsNumber());
+                System.out.println("Median of number of refactorings in commits " +
+                        "(only for commits that contain refactorings): " +
+                        executionInfo.getMedianOfNotNullRefactoringsNumbers());
+                System.out.println("Max number of refactorings in one commit: " +
+                        executionInfo.getMaxCommitRefactoringsNumber());
+                System.out.println("Detected: " + numberOfDetectedRefactorings);
+            } else {
+                System.out.println("Result: failed");
+                System.out.println("Error type: " +
+                        Objects.requireNonNull(repositoryDetectedRefactorings.getException()).getClass().getCanonicalName());
+                System.out.println("Reason: " +
+                        Objects.requireNonNull(repositoryDetectedRefactorings.getException()).getMessage());
+            }
+        }
+        System.out.println("--------------------------------");
+        RefactoringDetectionExecutionInfo totalExecutionInfo =
+                new RefactoringDetectionExecutionInfo(totalProcessedCommitsNumber, totalRefactoringsNumbersInProcessedCommits);
```

Comment:
It's better to create class which represents a compound of a few `RefactoringDetectionExecutionInfo` and put all the calculations inside of it.

Reply:
Ok. Combiners created.